src/porepy/models/constitutive_laws.py

```diff
@@ -2554,6 +2599,64 @@ def biot_stabilization(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
         return stabilization
 
 
+class BiotPoromechanicsPorosity(PoroMechanicsPorosity):
+    """Porosity for poromechanical models following classical Biot's theory.
+
+    The porosity is defined such that, after the chain rule is applied to the
+    accumulation term, the classical conservation equation from the Biot
+    equations is recovered.
+
+    Note that we assume constant fluid density and constant specific storage.
+
+    """
+
+    specific_storage: Callable[[list[pp.Grid]], pp.ad.Operator]
+    """Specific storage. Normally defined in a mixin instance of
+    :class:`~porepy.models.constitutive_laws.LinearElasticSolid`.
+
+    """
+
+    perturbation_from_reference: Callable[[str, list[pp.Grid]], pp.ad.Operator]
+    """Function that returns a perturbation from reference state. Normally provided by
+    a mixin of instance :class:`~porepy.models.VariableMixin`.
+
+    """
+
+    biot_coefficient: Callable[[list[pp.Grid]], pp.ad.Operator]
+    """Biot coefficient. Normally defined in a mixin instance of
+    :class:`~porepy.models.constitutive_laws.BiotCoefficient`.
+
+    """
+
+    def matrix_porosity(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
```

Comment:
This seems to contain quite a bit of duplication from base class. Could we avoid by some refactoring?

Reply:
Yes, this is true. I considered inheriting the value of `phi` from the base class, subtract the second term, and add the one with  the specific storage. But it turned out to be too confusing. I would rather not sacrifice the readability of the implementation for some lines of duplicated code.  Do you agree?

Reply:
Yes, but keeping readability should be acievable by refactoring matrix_porosity, I think.

Reply:
Oh. I misunderstood. If `matrix_porosity` is refactored we can definitely improved readability. I will give it a go.

Reply:
Done. Much better now.

Reply:
Done.